da/avail/avail.go

```diff
@@ -0,0 +1,192 @@
+package avail
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	ds "github.com/ipfs/go-datastore"
+	"github.com/rollkit/rollkit/da"
+	"github.com/rollkit/rollkit/da/avail/datasubmit"
+	"github.com/rollkit/rollkit/log"
+	"github.com/rollkit/rollkit/types"
+)
+
+type Config struct {
+	BaseURL    string  `json:"base_url"`
+	Seed       string  `json:"seed"`
+	ApiURL     string  `json:"api_url"`
+	AppID      int     `json:"app_id"`
+	Confidence float64 `json:"confidence"`
+}
+
+type DataAvailabilityLayerClient struct {
+	_      types.NamespaceID
+	config Config
+	logger log.Logger
+}
+
+type Confidence struct {
+	Block                uint32  `json:"block"`
+	Confidence           float64 `json:"confidence"`
+	SerialisedConfidence *string `json:"serialised_confidence,omitempty"`
+}
+
+type AppData struct {
+	Block      uint32   `json:"block"`
+	Extrinsics []string `json:"extrinsics"`
+}
+
+var _ da.DataAvailabilityLayerClient = &DataAvailabilityLayerClient{}
+var _ da.BlockRetriever = &DataAvailabilityLayerClient{}
+
+// Init initializes DataAvailabilityLayerClient instance.
+func (c *DataAvailabilityLayerClient) Init(_ types.NamespaceID, config []byte, kvStore ds.Datastore, logger log.Logger) error {
+	c.logger = logger
+
+	if len(config) > 0 {
+		return json.Unmarshal(config, &c.config)
+	}
+
+	return nil
+}
+
+// Start prepares DataAvailabilityLayerClient to work.
+func (c *DataAvailabilityLayerClient) Start() error {
+
+	c.logger.Info("starting avail Data Availability Layer Client", "baseURL", c.config.ApiURL)
+
+	return nil
+}
+
+// Stop stops DataAvailabilityLayerClient.
+func (c *DataAvailabilityLayerClient) Stop() error {
+
+	c.logger.Info("stopping Avail Data Availability Layer Client")
+
+	return nil
+}
+
+// SubmitBlock submits a block to DA layer.
+func (c *DataAvailabilityLayerClient) SubmitBlocks(ctx context.Context, blocks []*types.Block) da.ResultSubmitBlocks {
+
+	for _, block := range blocks {
+		data, err := block.MarshalBinary()
+		if err != nil {
+			return da.ResultSubmitBlocks{
+				BaseResult: da.BaseResult{
+					Code:    da.StatusError,
+					Message: err.Error(),
+				},
+			}
+		}
+		err = datasubmit.SubmitData(c.config.ApiURL, c.config.Seed, c.config.AppID, data)
+
+		if err != nil {
+			return da.ResultSubmitBlocks{
+				BaseResult: da.BaseResult{
+					Code:    da.StatusError,
+					Message: err.Error(),
+				},
+			}
+		}
+	}
+
+	return da.ResultSubmitBlocks{
+		BaseResult: da.BaseResult{
+			Code:     da.StatusSuccess,
+			Message:  "data submitted successfully",
+			DAHeight: 1,
+		},
+	}
+
+}
+
+// RetrieveBlocks gets the block from DA layer.
+
+func (c *DataAvailabilityLayerClient) RetrieveBlocks(ctx context.Context, dataLayerHeight uint64) da.ResultRetrieveBlocks {
+
+	blocks := []*types.Block{}
+
+Loop:
+	blockNumber := dataLayerHeight
+
+	appDataURL := fmt.Sprintf(c.config.BaseURL+"/appdata/%d?decode=true", blockNumber)
```

Comment:
can we use the full url from config and just do fmt.Sprintf here
```suggestion
	appDataURL := fmt.Sprintf(c.config.BaseURL+c.config.AppDataUrl, blockNumber)
```